### PR TITLE
Law: compressibility factor via intermolecular potential

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+## in the virial expansion account for interactions between successively larger groups of molecules. Since
+## interactions between large numbers of molecules are rare, the virial equation is usually truncated at the third
+## term onwards. Under the assumption that only pair interactions are present, the compressibility factor can be
+## linked to the intermolecular force potential:
+
+# Law: Z = 1 + 2 * pi * N / V * Integral((1 - exp(-phi(r) / (k * T))) * r**2, (r, 0, infinity))
+## Z - [compressibility factor](../../definitions/compressibility_factor_is_deviation_from_ideal_gas.py)
+## N - number of particles
+## V - volume
+## r - radius
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py:26]: Radius of what?

File: symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+## phi(r) - intermolecular force potential as a function of radius
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py:27]: Shouldn't it depend on the distance between molecules?

File: symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+
+@validate_input(
+    number_of_particles_=number_of_particles,
+    volume_=volume,
+    sphere_radius_=radius,
+)
+@validate_output(compressibility_factor)
+def calculate_compressibility_factor(
+    number_of_particles_: int | float,  # float to allow numbers written in exponent notation
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py:78]: Keep it as `int`. One can convert math notation number to int manually.

File: symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py (hunk 2 above)
Comment (alesanter) [reply symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py:27]: Yes, you are right

File: symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+## k - Boltzmann constant
+## T - absolute temperature
+
+# Conditions
+## - The virial expansion is done up to the second virial coefficient inclusively.
+
+compressibility_factor = Symbol("compressibility_factor", dimensionless, positive=True)
+number_of_particles = Symbol("number_of_particles", dimensionless, integer=True, positive=True)
+volume = Symbol("volume", units.volume, positive=True)
+radius = Symbol("radius", units.length, positive=True)
```
Comment (blackyblack) [inline symplyphysics/laws/thermodynamics/compressibility_factor_via_intermolecular_force_potential.py:37]: In the description it is distance, not radius. Update naming accordingly, or justify why you replace distance with radius.

**Review by blackyblack — APPROVED** [review]

**Pull request merged**